src/templatizer/array-selector.html

```diff
@@ -158,58 +187,48 @@
       }
 
       _applySplices(splices) {
+        let toRemove = [];
         for (let i=0; i<splices.length; i++) {
+          let removed = []
           let s = splices[i];
-          for (let j=0; j<s.removed.length; j++) {
-            this.deselect(s.removed[j]);
-          }
-          if (this.multi) {
-            this._addRemoveMulti(s.index, s.removed.length, s.addedCount);
-          } else {
-            this._addRemoveSingle(s.index, s.removed.length, s.addedCount);
-          }
-        }
-      }
-
-      _addRemoveMulti(start, removed, added) {
-        const links = this.__dataLinkedPaths;
-        for (let sidx=0; sidx<this.selected.length; sidx++) {
-          let idx = parseInt(links['selected.' + sidx].slice('items.'.length));
-          if (idx >= start) {
-            if (idx >= (start + removed)) {
-              this.linkPaths('selected.' + sidx, 'items.' + (idx + added - removed));
+          let selected = new IntegerSet();
+          let sidx = 0;
+          this._selectedSet.forEach(idx => {
```

Comment:
Apparently, the new set is created here because adding to a set while inside `forEach` is not supported. If that's the confirmed, let's add a comment.